lib/ansible/module_utils/basic.py

```diff
@@ -398,12 +401,12 @@ def __init__(self, argument_spec, bypass_checks=False, no_log=False,
         self._verbosity = 0
         # May be used to set modifications to the environment for any
         # run_command invocation
-        self.run_command_environ_update = {}
-        self._clean = {}
+        self.run_command_environ_update: dict[str, str] = {}
+        self._clean: dict[t.Never, t.Never] | str | None = {}
```

Comment:
@webknjaz `t.Never` doesn't work for Python < 3.11:

> 05:08 ERROR: lib/ansible/module_utils/basic.py:405:0: name-defined: Name "t.Never" is not defined (100%)

(I'm not sure why mypy actually has to pass for all Python versions, wouldn't it be enough to run it for only the latest Python version, or just the controller Python versions? After all `from __future__ import annotations` is used everywhere.)
```suggestion
        self._clean: dict | str | None = {}
```